contrib/windows/cygwin.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+
+mirror=http://mirrors.mit.edu/cygwin
+platform=x86
+toinstall="mintty cygwin coreutils libintl8 libiconv2"
+local_path=./cygwin
+
+# make place to put downloads
+mkdir -p $local_path
+
+# get setup.ini
+curl $mirror/$platform/setup.bz2 > $local_path/setup.ini.bz2
+bunzip2 -f $local_path/setup.ini.bz2
+
+# get and unpack packages
+for f in $toinstall
+do
+    # determine the relative URL
+    url=$(awk "BEGIN {RS = \"@ \"} ; \$1 == \"$f\" { print \$0 }" $local_path/setup.ini | awk '$1 == "install:" { print $2 }' - | head -1)
+
+    # determine the local filename
+    filename=${url##*/}
```

Comment:
This looks like a bashism, so `#!/bin/bash` would be a more honest shebang. A `set -e` early on would also be a good idea.